File: create-icons.py
#!/usr/bin/env python3
"""
Script para gerar ícones PWA a partir do logo da empresa
Uso: python create-icons.py
Requer: Pillow (pip install Pillow)
"""

import os
import sys
from PIL import Image

def create_icons():
    """Gera ícones PWA a partir do logo"""
    
    # Caminhos
    public_dir = os.path.dirname(os.path.abspath(__file__)) + '/public'
    source_image = os.path.join(public_dir, 'Secontaf1.png')
    
    # Verificar se arquivo existe
    if not os.path.exists(source_image):
        print(f"❌ Arquivo não encontrado: {source_image}")
        sys.exit(1)
    
    # Tamanhos necessários
    sizes = [
        ('icon-192.png', 192),
        ('icon-512.png', 512),
        ('icon-192-maskable.png', 192),
        ('icon-512-maskable.png', 512),
    ]
    
    try:
        print("📦 Gerando ícones PWA...\n")
        
        # Abrir imagem original
        img = Image.open(source_image).convert('RGBA')
        
        for filename, size in sizes:
            output_path = os.path.join(public_dir, filename)
            
            print(f"  Criando {filename} ({size}x{size})...")
            
            # Criar ícone com fundo transparente
            icon = Image.new('RGBA', (size, size), (255, 255, 255, 0))
            
            # Redimensionar imagem mantendo proporção
            img_resized = img.copy()
            img_resized.thumbnail((size, size), Image.Resampling.LANCZOS)
            
            # Calcular posição para centralizar
            x = (size - img_resized.width) // 2
            y = (size - img_resized.height) // 2
            
            # Colar imagem no centro
            icon.paste(img_resized, (x, y), img_resized)
            
            # Salvar
            icon.save(output_path, 'PNG')
            print(f"  ✓ {filename} criado com sucesso")
        
        print("\n✅ Todos os ícones foram criados com sucesso!\n")
        print("Arquivos gerados:")
        for filename, _ in sizes:
            print(f"  ✓ public/{filename}")
        print("\nOs ícones estão prontos para usar no manifest.json!")
        
    except ImportError:
        print("❌ Erro: Pillow não está instalado.")
        print("\nPara usar este script, instale Pillow:")
        print("  pip install Pillow")
        print("\nOu use uma ferramenta online:")
        print("  https://www.pwabuilder.com/")
        sys.exit(1)
    except Exception as e:
        print(f"❌ Erro ao gerar ícones: {e}")
        sys.exit(1)

if __name__ == '__main__':
    create_icons()

File: create-icons.js
/**
 * Script para gerar ícones PWA a partir do logo da empresa
 * Execute: node create-icons.js
 */

import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));

// Tentar usar sharp se disponível, senão usar canvas
let sharp;
try {
  sharp = await import('sharp').then(m => m.default);
} catch (e) {
  console.warn('Sharp não instalado. Tentando usar método alternativo...');
}

const publicDir = path.join(__dirname, 'public');
const sourceImage = path.join(publicDir, 'Secontaf1.png');

if (!fs.existsSync(sourceImage)) {
  console.error('❌ Arquivo não encontrado:', sourceImage);
  process.exit(1);
}

// Tamanhos necessários para PWA
const sizes = [
  { name: 'icon-192.png', size: 192 },
  { name: 'icon-512.png', size: 512 },
  { name: 'icon-192-maskable.png', size: 192 },
  { name: 'icon-512-maskable.png', size: 512 },
];

async function createIcons() {
  if (!sharp) {
    console.error('❌ Erro: Sharp não está instalado.');
    console.log('\nPara usar este script, instale sharp:');
    console.log('  npm install --save-dev sharp');
    console.log('\nOu use uma ferramenta online:');
    console.log('  https://www.pwabuilder.com/');
    process.exit(1);
  }

  console.log('📦 Gerando ícones PWA...\n');

  try {
    for (const icon of sizes) {
      const outputPath = path.join(publicDir, icon.name);
      
      console.log(`  Criando ${icon.name} (${icon.size}x${icon.size})...`);
      
      await sharp(sourceImage)
        .resize(icon.size, icon.size, {
          fit: 'contain',
          background: { r: 255, g: 255, b: 255, alpha: 0 } // Fundo transparente
        })
        .png()
        .toFile(outputPath);

      console.log(`  ✓ ${icon.name} criado com sucesso`);
    }

    console.log('\n✅ Todos os ícones foram criados com sucesso!\n');
    console.log('Arquivos gerados:');
    sizes.forEach(icon => {
      console.log(`  ✓ public/${icon.name}`);
    });
    console.log('\nOs ícones estão prontos para usar no manifest.json!');

  } catch (error) {
    console.error('❌ Erro ao gerar ícones:', error.message);
    process.exit(1);
  }
}

createIcons();
